lib/world.rb

```diff
@@ -58,35 +65,32 @@ def neighbors_of(x, y)
     else  
       neighbors.push(@cells[x + 1][y + 1]) # Bottom Right
     end
-
     if x == @rows - 1
       neighbors.push(@cells[0][y - 1]) # Wrap from bottom to top
     else
       neighbors.push(@cells[x + 1][y - 1]) # Bottom left
     end
-
     neighbors.push(@cells[x - 1][y]) # Top
-
     if y == @columns - 1
       neighbors.push(@cells[x - 1][0]) # Wrap from right to left
     else
       neighbors.push(@cells[x - 1][y + 1]) # Top Right
     end
-
     neighbors.push(@cells[x - 1][y - 1]) # Top Left
+    return neighbors
   end
 
   def alive_neighbors_of(x, y)
     alive_neighbors = []
-    neighbors_of(x, y).each do |neighbor|
-      if neighbor == "x"
+    neighbors = neighbors_of(x, y)
+
+    neighbors.each do |neighbor|
```

Comment:
Now that you have a message you can send a Cell to tell whether it's alive, you are able to simplify this loop using some more advanced Ruby.

```
alive_neighbors = neighbors.filter { |neighbor| neighbor.alive? }
```

Reply:
New efficient code structures excite me